### PR TITLE
Add hpc-workspace to OpenHPC

### PR DESCRIPTION
**Pull request by georgiastuart**
This is an attempt to fulfill #1403 

[hpc-workspace](https://github.com/holgerBerger/hpc-workspace) is an admin tool for managing directories with associated expiration dates and share groups. 

This PR adds:
- hpc-workspace spec file
- hpc-workspace license
- hpc-workspace config template

Open questions:
- Is the admin category the correct spot for this component?
- I included the default daily cleanup script, should that be changed?

### CODE REVIEW

File: components/admin/hpc-workspace/SPECS/hpcworkspace.spec [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+working filesystems in an HPC environment.
+
+%prep
+%setup -q -n %{pname}-%{version}
+
+%build
+cmake \
+  -DCMAKE_INSTALL_PREFIX=%{install_path} \
+  .
+make -j 4
```
Comment (adrianreber) [inline components/admin/hpc-workspace/SPECS/hpcworkspace.spec]: Please do not hard code the number of parallel tasks to 4. There is a macro for this.

File: components/admin/hpc-workspace/SPECS/hpcworkspace.spec [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+%{_sysconfdir}/cron.daily/%{pname}
+%attr(4755, root, root) %{install_path}/bin/ws_allocate
+%attr(4755, root, root) %{install_path}/bin/ws_release
+%attr(4755, root, root) %{install_path}/bin/ws_restore
+
+
+%pre
+# provide specific uid/gid to ensure that it is the same across the cluster
+/usr/bin/getent group hpcws >/dev/null 2>&1 || \
+  /usr/sbin/groupadd -r hpcws -g 85
```
Comment (adrianreber) [inline components/admin/hpc-workspace/SPECS/hpcworkspace.spec]: I am curious about the 85. Not sure we have a central place in openhpc where we decide on pre defined UIDs and GIDs. Just curious why the 85.

**Comment by adrianreber** [comment]
Please have a look at the result of the CI run on Leap. There seems to be an error.

File: components/admin/hpc-workspace/SPECS/hpcworkspace.spec (hunk 2 above)
Comment (georgiastuart) [reply components/admin/hpc-workspace/SPECS/hpcworkspace.spec]: Looks like munge is 201 and slurm is 202. Would it be better to change it to 203?

**Comment by adrianreber** [comment]
> This is an attempt to fulfill #1403 

Thanks!

> Open questions:
> - Is the admin category the correct spot for this component?

Sounds good to me.

> - I included the default daily cleanup script, should that be changed?

That is indeed a difficult question. On the one hand I think this is the correct solution, on the other hand it could be a site specific setup. I would leave it as it is. Make sure it is mentioned in the documentation. Also maybe the cron setup should be marked as a config file so that RPM does not overwrite local changes on upgrade.

File: components/admin/hpc-workspace/SPECS/hpcworkspace.spec (hunk 2 above)
Comment (adrianreber) [reply components/admin/hpc-workspace/SPECS/hpcworkspace.spec]: I have no idea. I was just curious. We probably need the input from @koomie. Maybe he remembers how we distribute static UIDs in Openhpc.

File: components/admin/hpc-workspace/SPECS/hpcworkspace.spec (hunk 1 above)
Comment (georgiastuart) [reply components/admin/hpc-workspace/SPECS/hpcworkspace.spec]: Resolved in ca83914

**Comment by georgiastuart** [comment]
Open tasks:
- [ ] Mark cron job as config in spec
- [ ] Add hpc-workspace-ohpc to documentation files
- [x] Resolve UID/GID numbering

**Comment by ohpcsim** [comment]
Hey @georgiastuart,

For the cron job, we've decided not to include it as part of the RPM. We will document what needs to be done in the recipes and let sysadmins decide when to run it themselves based on their local usage patterns.

For the UID/GID numbering, you can go ahead and remove the specific group choice (85) and we will let the order of operations decide the UID/GID numbering as we do for SLURM/PBS and Munge.

I'll go ahead and do the documentation as well. If you'd like to make the above changes, and resubmit great. If you don't have the time, let me know and I'll do it.

Thanks again for your contribution!

**Comment by georgiastuart** [comment]
@mghpcsim Ready for the next review. 

- uid/gid is now 203 in spec and config template per our discussion 
- cron job removed

**Comment by georgiastuart** [comment]
Investigating the test failure.

**Pull request merged**